server/src/main/java/org/opensearch/search/aggregations/InternalAggregation.java

```diff
@@ -160,6 +161,14 @@ public boolean isSliceLevel() {
             return this.isSliceLevel;
         }
 
+        public int getRequiredSizeLocal(TermsAggregator.BucketCountThresholds bucketCountThresholds) {
+            return isSliceLevel() ? bucketCountThresholds.getShardSize() : bucketCountThresholds.getRequiredSize();
+        }
+
+        public long getMinDocCountLocal(TermsAggregator.BucketCountThresholds bucketCountThresholds) {
+            return isSliceLevel() ? bucketCountThresholds.getShardMinDocCount() : bucketCountThresholds.getMinDocCount();
```

Comment:
Lets add a comment to these methods:

``` 
For slice level partial reduce we will apply shard level `size` and `minDocCount` limits whereas for coordinator level partial reduce it will use top level `size` and `minDocCount`